gstools/transform/field.py

```diff
@@ -67,6 +69,56 @@ def binary(fld, divide=None, upper=None, lower=None):
         fld.field[fld.field <= divide] = lower
 
 
+def discrete(fld, values, thresholds=None):
+    """
+    Discrete transformation.
+
+    After this transformation, the field has only `len(values)` discrete
+    values.
+
+    Parameters
+    ----------
+    fld : :any:`Field`
+        Spatial Random Field class containing a generated field.
+        Field will be transformed inplace.
+    values : :any:`np.ndarray`
+        The discrete values the field will take
+    thresholds : :class:`numpy.ndarray`, optional
+        the thresholds, where the value classes are separated
+        Default: mean of the neighbouring values
+    """
+    if fld.field is None:
+        print("discrete: no field stored in SRF class.")
+    else:
+        if thresholds is None:
+            # just in case, sort the values
+            values = np.sort(values)
+            thresholds = (values[1:] + values[:-1]) / 2
```

Comment:
I think the natural way would be to us thresholds, so that the resulting ratios between the given values are even. Maybe I don't get the aim of this transformation...

Reply:
Yeah, I was also not exactly sure how to handle the kind of input arguments the best.
But at least in my use case I'm just interested in the number of "value classes". The non-arithmetic mean thresholds are just a little generalisation, which was easy to implement.

If the function would just take an array of thresholds, what values would you assign to all the field values lying in between two thresholds?